tests/unit/components/ObsEdit/ObsEdit.partIntegration.test.js

```diff
@@ -0,0 +1,147 @@
+// without any current observation or ability to update or fetch
+// observations
+const mockObsEditProviderWithObs = obs => ObsEditProvider.mockImplementation( ( { children } ) => (
+  // eslint-disable-next-line react/jsx-no-constructed-context-values
+  <ObsEditContext.Provider value={{
+    observations: obs,
+    currentObservation: obs[0]
+  }}
+  >
+    {children}
+  </ObsEditContext.Provider>
+) );
+
+const renderObsEdit = ( ) => renderComponent(
+  <ObsEditProvider>
+    <ObsEdit />
+  </ObsEditProvider>
+);
+
+test( "renders observation photo from photo gallery", ( ) => {
+  const observations = [factory( "RemoteObservation", {
+    latitude: 37.99,
+    longitude: -142.88,
+    user: mockCurrentUser,
+    place_guess: mockLocationName
+  } )];
+  mockObsEditProviderWithObs( observations );
+
+  const { getByText } = renderObsEdit( );
+
+  const obs = observations[0];
+
+  expect( getByText( obs.place_guess ) ).toBeTruthy( );
+  expect( getByText( new RegExp( obs.longitude ) ) ).toBeTruthy( );
+} );
+
+describe( "location fetching", () => {
+  beforeEach( () => {
+    // resets mock back to original state
+    mockFetchUserLocation.mockReset();
+  } );
+  test( "should fetch location when new observation hasn't saved", async ( ) => {
+    const observations = [{}];
+    mockObsEditProviderWithObs( observations );
+    expect( mockFetchUserLocation ).not.toHaveBeenCalled();
+
+    renderObsEdit( );
+
+    await waitFor( () => {
+      expect( mockFetchUserLocation ).toHaveBeenCalled();
+    } );
+    // Note: it would be nice to look for an update in the UI, but since we've
+    // mocked ObsEditProvider here, it will never update. Might be good for
+    // an integration test
+  } );
+
+  test( "shouldn't fetch location for existing obs on device that hasn't uploaded", async ( ) => {
+    const observation = factory( "LocalObservation" );
+    expect( observation.id ).toBeFalsy( );
+    expect( observation.created_at ).toBeFalsy( );
+    expect( observation._created_at ).toBeTruthy( );
+    mockObsEditProviderWithObs( [observation] );
+    const { queryByText } = renderObsEdit( );
+
+    expect( queryByText( new RegExp( `Lat: ${observation.latitude}` ) ) ).toBeTruthy( );
+    expect( mockFetchUserLocation ).not.toHaveBeenCalled();
+  } );
+
+  test( "shouldn't fetch location for existing observation created elsewhere", async ( ) => {
+    const observation = factory( "LocalObservation", {
+      id: faker.datatype.number( ),
+      created_at: faker.date.past( ),
+      _synced_at: faker.date.past( )
+    } );
+    expect( observation.id ).toBeTruthy( );
+    expect( observation.created_at ).toBeTruthy( );
+    mockObsEditProviderWithObs( [observation] );
+    const { queryByText } = renderObsEdit( );
+
+    expect( queryByText( new RegExp( `Lat: ${observation.latitude}` ) ) ).toBeTruthy( );
+    expect( mockFetchUserLocation ).not.toHaveBeenCalled();
+  } );
+} );
```

Comment:
What is a `partIntegration` test? Shouldn't tests that address interaction between components and other systems just be in an integration test?

Reply:
Nothing really. I renamed and moved it into the integration folder.